install_config/persistent_storage/persistent_storage_glusterfs.adoc

```diff
@@ -13,567 +14,208 @@ toc::[]
-- A distinct list of servers (IP addresses) in the Gluster cluster, to be defined as endpoints.
-- A service, to persist the endpoints (optional).
-- An existing Gluster volume to be referenced in the persistent volume object.
-- *glusterfs-fuse* installed on each schedulable {product-title} node in your cluster:
+include::install_config/persistent_storage/topics/glusterfs_support_requirements.adoc[]
+
+endif::[]
+
+[[install]]
+== Installation
+
+For standalone {gluster}, there is no component installation required to use it
+with {product-title}. {product-title} comes with a built-in GlusterFS volume
+driver, allowing it to make use of existing volumes on existing clusters. See
+xref:provisioning[provisioning] for more on how to make use of existing volumes.
+
+For {gluster-native} and {gluster-external}, it is recommended to use the
+xref:../../install/advanced_install.adoc[Advanced Installer] to install the
+required components.
+
+[[install-gluster-nodes]]
+=== {gluster-external}: Installing {gluster} Nodes
+
+include::install_config/persistent_storage/topics/glusterfs_install_nodes.adoc[]
+
+[[install-advanced-installer]]
+=== Using the Advanced Installer
+
+The xref:../../install/advanced_install.adoc[Advanced Installer] can be used to
+install one or both of two GlusterFS clusters:
+
+- `glusterfs`: A general storage cluster for use by user applications.
+- `glusterfs-registry`: A dedicated storage cluster for use by infrastructure
+applications such as an integrated OpenShift Container Registry.
+
+Both of these are defined in the inventory hosts file.
+
+The definition of the clusters is done by including the relevant names in the
+`[OSEv3:children]` group, creating similarly named groups, and then populating
+the groups with the node information. The clusters can then be configured
+through a variety of variables in the `[OSEv3:vars]` group. `glusterfs`
+variables begin with `openshift_storage_glusterfs_` and `glusterfs-registry`
+variables begin with `openshift_storage_glusterfs_registry_`. A few other
+variables, such as `openshift_hosted_registry_storage_kind`, interact with the
+GlusterFS clusters.
+
+It is recommended to specify version tags for all containerized components.
+This is primarily to prevent components, particularly the {gluster} pods, from
+upgrading after an outage which may lead to a cluster of widely disparate
+software versions. The relevant variables are:
+
+- `openshift_storage_glusterfs_version`
+- `openshift_storage_glusterfs_block_version`
+- `openshift_storage_glusterfs_s3_version`
+- `openshift_storage_glusterfs_heketi_version`
+- `openshift_storage_glusterfs_registry_version`
+- `openshift_storage_glusterfs_registry_block_version`
+- `openshift_storage_glusterfs_registry_s3_version`
+- `openshift_storage_glusterfs_registry_heketi_version`
+
+For a complete list of variables, see the
+link:{gluster-role-link}[GlusterFS role README] on GitHub.
```

Comment:
I find the general lack of any in-depth documentation for openshift-ansible in openshift-docs to be a severe shortcoming, even more so when an attempt of that documentation is made but then falls out of sync with what's actually in the product. Both of these have caused me quite some pain in the past.

Based on this, I will insist this stay and be published until the broader systemic problem can be improved and a better alternative can be found. If your concern is primarily with the openshift-enterprise documentation, I would be willing to put a conditional around the link so that it only shows up for openshift-origin.

Reply:
Please